repository/src/main/java/com/epam/esm/repository/impl/GiftCertificatePreparedStatementCreator.java

```diff
@@ -0,0 +1,96 @@
+package com.epam.esm.repository.impl;
+
+import org.springframework.jdbc.core.PreparedStatementCreator;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class GiftCertificatePreparedStatementCreator implements PreparedStatementCreator {
+
+    private String tag;
+    private String search;
+    private final Map<SortColumn, SortType> sorts = new LinkedHashMap<>();
+    private String queryString;
+
+    @Override
+    public PreparedStatement createPreparedStatement(Connection connection) throws SQLException {
+            PreparedStatement preparedStatement = connection.prepareStatement(queryString);
+            int counter = 1;
+            if (tag != null) {
+                preparedStatement.setString(counter++, tag);
+            }
+
+            if (search != null) {
+                preparedStatement.setString(counter++, search);
+                preparedStatement.setString(counter, search);
+            }
+            return preparedStatement;
+    }
+
+    public static GiftCertificateQueryBuilder createBuilder() {
+        return new GiftCertificatePreparedStatementCreator().new GiftCertificateQueryBuilder();
+    }
+
+    public class GiftCertificateQueryBuilder {
```

Comment:
Do you need this class to be public?

Reply:
If it is private I will not be able to use it is service